### ISSUE TITLE
Sort "by status" puts "at most" habits in the wrong place

### ISSUE THREAD

**Issue by hiqua** [iSoron/uhabits#1508]
## Pre-submission checklist
- [X] I am submitting a bug report, not a feature request.
- [X] I have searched for similar issues, but did not find any matches.

## Description
A clear and concise description of what the problem is.

## Steps to reproduce
1. create several habits with various goals
2. tick them differently
3. sort by status

You'd expect an AT MOST habit that's still unfilled for the day to appear somewhat higher, right now it appears after habits you've ticked for the day.

**Comment by iSoron** [comment]
@hiqua I'm actually not entirely sure about this one; I think it makes some sense for zero'ed AT MOST habits to appear together with completed ones. When scanning the list sorted by status, I imagine that users are looking for things they should do next. A zero'ed AT MOST habit is already at its perfect state, so there's no action needed, just like completed regular habits. I imagine you would typically enter values for AT MOST habits during the day, when you regrettably perform the habit; in that case, its position in the list is not that important, since you opened the app looking specifically for that habit.

Does this make any sense? Or is this issue specifically about users who have "show question marks" enabled?

**Comment by hiqua** [comment]
@iSoron yes sorry it's about the question marks, where an action is somewhat expected even if to say "not done" or "0" for the AT_MOST habits.

**Comment by Shyam-NR** [comment]
Hey I have a quick question... I am tracking this to quit a habit and I have set ATMOST to 0. 
So shouldn't the score go towards 100% when I reach towards my ATMOST variable?
But it isn't the case now
